Add documentation for cli-generate:artifacts DevTool (#743)
Enalean/tuleap-documentation-en: languages/en/developer-guide/development-tools.rst

Add a break point (in pre.php for instance)
Go on your tuleap page, then click on bookmark then refresh your page,
debugger should start

Generate artifacts with random data
-----------------------------------

Use *cli-generate:artifacts* CLI development tool to generate 1, 10, 1 000 000 artifacts in XML format.

The command line will generate an artifacts.xml with the given number of artifacts containing random:
 * number of changesets
 * fields
 * field values

**1. Download the sources from Tuleap Git**

SSH: ``git clone ssh://gitolite@tuleap.net/service-cleanup/cli-generate-stuffs.git``
    HTTPS: ``git clone https://tuleap.net/plugins/git/service-cleanup/cli-generate-stuffs.git``

**2. Installation**

.. attention::

>     Your tuleap instance has to be running to be able to use this DevTool.

Author: Yes indeed, thanks for making me notice that!

\

 * Run ``make composer`` to install project dependencies
 * Run ``make setup`` to retrieve the certificate of your Tuleap development instance
 * Add a config.json file in root directory to store your API access key as json data: ``{'access_key': 'youraccesskey' }``

**3. How to use it?**

Run ``bin/console generate:artifacts [number_of_artifacts] [tracker id] [opt: title for all artifacts with semantic title]``

>     example: ``bin/console generate:artifacts 100 1`` will generate 100 artifacts with the structure of the tracker with id 1.

Author: :+1:

> 

Author: You're right for the data directory in root directory: it contains generated random artifact files.
Instead, the XML artifacts will be generated in a single file: artifacts.xml directly in root directory.

**4. Import the generated XML artifacts in a project**

> 

Reviewer: Maybe add a bullet point to explain that we need an archive exported by Tuleap, (with a link to export XML doc), and we need to unzip it.
Because at this moment, we don't know that we need it

Author: :+1:

>  * Copy/paste the ``<artifacts>`` node into the matching node of your ``project.xml`` file.

Reviewer: What do you think to add `[...] of your project.xml file exported to the previous step`? It will be more understandable

Author: :+1:

 * In console mode, from root directory: ``cp data/* {your-project-directory}/data``

>  * Use Tuleap CLI importer with that compressed project directory.

Author: :+1:

Go on your tuleap page, then click on bookmark then refresh your page,
debugger should start

Generate artifacts with random data
-----------------------------------

Use *cli-generate:artifacts* CLI development tool to generate 1, 10, 1 000 000 artifacts in XML format in order to test code performance.

The command line will generate an artifacts.xml file with the given number of artifacts containing random:
 * number of changesets
 * fields
 * field values

**1. Download the sources from Tuleap Git**

* SSH: ``git clone ssh://gitolite@tuleap.net/service-cleanup/cli-generate-stuffs.git``
* HTTPS: ``git clone https://tuleap.net/plugins/git/service-cleanup/cli-generate-stuffs.git``

**2. Installation**

.. attention::

    Your tuleap instance has to be running to be able to use this DevTool.

* Make sure your tuleap instance is running on this URL: https://tuleap-web.tuleap-aio-dev.docker
* Run ``make composer`` to install project dependencies
* Run ``make setup`` to retrieve the certificate of your Tuleap development instance
* Add a config.json file in root directory to store your API access key as json data: ``{'access_key': 'youraccesskey' }``

**3. How to use it?**

> 

Reviewer: Please could you keep the old line "Run `bin/console generate:artifacts [number_artifacts] [tracker_id] [OPT: title]`".
That helped to understand which option we have with this CLI.
Maybe instead `Run`, you can replace it by `There are 3 parameters for the CLI: ...`

For example, the following line will generate 100 artifacts with the structure of the tracker with id 1 and with the same (optional) title "random":

.. sourcecode:: shell

    $> bin/console generate:artifacts 100 1 "random"

> The previous command line will generate artifacts in an artifacts.xml file at root directory and random artifact files in ``./data``.

Reviewer: ```suggestion
The previous command line will generate a directory that contains an XML file named `artifacts.xml` and a directory ``./data`` that contain artifact files within.
```
What do you think?

**4. Export the project in which you want to inject the XML artifacts node** \

* Please follow `Project Export documentation <https://docs.tuleap.org/administration-guide/projects-management/export-import/project-export.html>`_
* Unzip the project directory you've just exported

**5. Import the generated XML artifacts in a project**

* Copy/paste the ``<artifacts>`` node into the matching node of your ``project.xml`` file exported to the previous step

> * In console mode, from root directory, copy the contents of data directory into your data project directory: ``cp data/* {your-project-directory}/data``

Reviewer: What do you think to replace {your-project-directory} by {exported-project-directory}? And use `terminal` instead `console mode`?

```suggestion
* In terminal, from root directory, copy the contents of data directory into your data project directory: ``cp data/* {exported-project-directory}/data``
```

Author: :+1:
